src/shared/telemetry/spans.ts

```diff
@@ -162,6 +163,7 @@ export class TelemetrySpan {
 interface TelemetryContext {
     readonly spans: TelemetrySpan[]
     readonly activeSpan?: TelemetrySpan
+    readonly attributes?: Partial<MetricShapes[MetricName]>
```

Comment:
Something I've been meaning to add. I came across a use-case here so I just added it now.